src/message-backup/BackupClientFactory.ts

```diff
@@ -0,0 +1,53 @@
+import BackupClient, {
+  BackupConfiguration,
+  BackupType,
+  SelectBackupProvider,
+} from './BackupClient'
+import NoBackupClient from './NoBackupClient'
+import XMTPBackupClient from './XMTPBackupClient'
+
+/**
+ * Creates a backup client of the correct provider type (e.g. xmtp backup, no backup, etc).
+ * Uses an existing user preference from the backend if it exists, else prompts for a new
+ * one using the `providerSelector`
+ *
+ * @param walletAddress The public address of the user's wallet
+ * @param selectBackupProvider A callback for determining the provider to use, in the event there is no
+ * existing user preference. The app can define the policy to use here (e.g. prompt the user,
+ * or default to a certain provider type).
+ * @returns A backup client of the correct type
+ */
+export async function createBackupClient(
+  walletAddress: string,
+  selectBackupProvider: SelectBackupProvider
+): Promise<BackupClient> {
+  const configuration = await fetchOrCreateConfiguration(
+    walletAddress,
+    selectBackupProvider
+  )
+  switch (configuration.provider.type) {
+    case BackupType.none:
+      return new NoBackupClient(configuration)
+    case BackupType.xmtp:
+      return new XMTPBackupClient(configuration)
```

Comment:
What is the significance of XTMP in the name XMTPBackupClient. What other BackupClients do you imagine in the future?